### PR TITLE
KEYCLOAK-7594 Upgrade to Wildfly 13, Upgrade JDG for cross-dc to 7.2

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

File: pom.xml
Comment (mingjliu9) [inline pom.xml]: If my understanding is right, we no longer need this `org.jboss.aesh` dependency now.

Comment (mposolda) [reply pom.xml]: @mingjliu9 It's unfortunately still needed as keycloak-admin-cli and keycloak-client-registration-cli are still using it. Both CLIs are packed in server distribution in directory KEYCLOAK_HOME/bin/client/ in the "fat" jars, which among other things, contain the classes of old aesh. I've created https://issues.jboss.org/browse/KEYCLOAK-7737 . 

Fortunately org.jboss.aesh is not needed in jboss modules, so IMO KEYCLOAK-7737 is not a blocker for this PR, but can be done later.

File: adapters/oidc/js/src/main/resources/keycloak.d.ts
Comment (stianst) [inline adapters/oidc/js/src/main/resources/keycloak.d.ts]: What's happening with this file? Some sort of encoding change?

File: distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli
Comment (stianst) [inline distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli]: What's with the empty echo statements here?

File: pom.xml
Comment (stianst) [inline pom.xml]: This needs to be changed to corresponding EAP CD build. That's 7.2.0.CD13-redhat-3. Also need to check that RH-SSO works as expected.

File: pom.xml
Comment (stianst) [inline pom.xml]: This also needs to be updated

**Review by stianst — CHANGES_REQUESTED** [review]
Some changes needed, specifically changing EAP version for product profile so RH-SSO CD3 can be built. We also need to test that RH-SSO works.

We should probably run a full CI matrix on both Keycloak and RH-SSO before merging this as well.

File: distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli
Comment (mposolda) [reply distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli]: That's just for adding another line to the output of the CLI command. I've just tried without echo and the output doesn't look so nice. Also the pattern with empty echo is used in whole file and all the previous instructions.

File: adapters/oidc/js/src/main/resources/keycloak.d.ts
Comment (mposolda) [reply adapters/oidc/js/src/main/resources/keycloak.d.ts]: Yes, it seems that my linux/github changed the line encodings of this file from LF to CRLF and seem it's doing it automatically. Will need to look at it to avoid this change...

File: pom.xml
Comment (mposolda) [reply pom.xml]: Ok, will look at it

File: distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli
Comment (mposolda) [reply distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli]: Just to compare, here is the output file with empty echo command:
``
Removing eviction from keycloak users cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}

Removing eviction from keycloak authorization cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}

Removing eviction from keycloak keys cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}

``

And here the one without echo:
``
Removing eviction from keycloak users cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
Removing eviction from keycloak authorization cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
Removing eviction from keycloak keys cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
``

My vote is, that the first one looks better

**Comment by mposolda** [comment]
@stianst Thanks for the review. Will try to look at keycloak.d.ts file and updating EAP version and running CI matrix for both Keycloak and RH-SSO.

**Review by vramik — COMMENTED** [review]
ConsoleProtectionTest fails with Wildfly13. There might been some changes since WF11 .

**Review by hmlnarik — COMMENTED** [review]
@mposolda Nice, thanks! I need to do play a bit with it but overall looks good to me.

**Comment by mposolda** [comment]
@stianst Addressed your suggestions (despite empty echo statements in the migration scripts)

@hmlnarik I've figured that in the Keycloak distribution, we already removed the headers like "Server" and "X-Powered-By" . I think this was intentional to not reveal the things like version etc. So I've just removed it from this PR.

@vramik The ConsoleProtectionTest is still failing (sent mail to keycloak-team for decide how to move forward).

File: testsuite/integration-arquillian/servers/app-server/jboss/eap/src/main/java/org/keycloak/testsuite/arquillian/eap/container/EAPAppServerProvider.java
Comment (vramik) [inline testsuite/integration-arquillian/servers/app-server/jboss/eap/src/main/java/org/keycloak/testsuite/arquillian/eap/container/EAPAppServerProvider.java]: Wouldn't be better to also test that property is not null or empty? I know it is set by property in pom.xml but in case someone accidentally removes it it will start failing with NPE. See the lines below.

**Review by hmlnarik — COMMENTED** [review]
@mposolda Thanks!
My concerns have been addressed, lgtm.

File: testsuite/integration-arquillian/servers/app-server/jboss/eap/src/main/java/org/keycloak/testsuite/arquillian/eap/container/EAPAppServerProvider.java
Comment (mposolda) [reply testsuite/integration-arquillian/servers/app-server/jboss/eap/src/main/java/org/keycloak/testsuite/arquillian/eap/container/EAPAppServerProvider.java]: @vramik Sorry I missed this validation. Added: Validate.notNullOrEmpty(containerName, "app.server is not set.");

**Review by vramik — APPROVED** [review]
@mposolda thank you, lgtm

**Review by stianst — APPROVED** [review]

**Pull request merged**
